### PR TITLE
Added support for Google Cloud.

### PR DESCRIPTION
**Pull request by narmaku**

### CODE REVIEW

File: cloud/terraform/terraform_controller.py [hunk 1]
```diff
@@ -110,6 +109,28 @@ def get_instances_azure(self, resources):
...
+            if resource['type'] != 'google_compute_instance':
+                continue
+
+            public_ip = resource['values']['network_interface'][0]['access_config'][0]['nat_ip']
+
+            instances_info[resource['address']] = {
+                'name': resource['name'],
+                'instance_id': resource['values']['id'],
+                'public_ip': public_ip,
+                'public_dns': public_ip,
```
Comment (jabia99) [inline cloud/terraform/terraform_controller.py:126]: No DNS for google images? interesting

File: cloud/terraform/gcloud_config_builder.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        new_vpc = {
+            'name': network_name,
+            'auto_create_subnetworks': True
+        }
+
+        self.resources_tf['resource']['google_compute_network'][network_name] = new_vpc
+
+    def __new_gcloud_instance(self, instance):
+        if not instance['instance_type']:
+            instance['instance_type'] = 'e2-micro'
```
Comment (jabia99) [inline cloud/terraform/gcloud_config_builder.py:87]: Are we sure this instance type is enough for our testing? Maybe we should add a TODO so we don't forget to verify it

File: cloud/terraform/gcloud_config_builder.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            'tags': [self.ssh_enabled_tag, self.ci_tag_key],
+            'depends_on': [
+                'google_compute_network.{}'.format(instance['google_compute_network'])
+            ]
+        }
+
+        self.resources_tf['resource']['google_compute_instance'][name] = new_instance
+
+    def __get_gcloud_resource_self_link(self, resource_name):
+        pass
```
Comment (jabia99) [inline cloud/terraform/gcloud_config_builder.py:135]: this is still WIP, right?

**Review by jabia99 — COMMENTED** [review]
Just a few questions

File: cloud/terraform/gcloud_config_builder.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                'image': instance['image']
+            }
+        }
+
+        network_interface = {
+            'network': instance['google_compute_network'],
+            'access_config': {}
+        }
+
+        if 'username' in instance:
```
[the remark this answers is not in the document]
Comment (narmaku) [reply cloud/terraform/gcloud_config_builder.py:108]: For Azure and GCP it is not needed to specify a username (unlike AWS, which the user name to be used is the one already "embedded" into the AMI). So, in AZ and GCP you can use whatever username you like, but in AWS it has to be yes or yes the one defined inside the AMI.

In Azure we require "username" field to be present in the resources.json but we shouldn't (this is an improvement we have to do). So, in GCP I directly made it optional, just like it should be.

File: cloud/terraform/gcloud_config_builder.py (hunk 2 above)
Comment (narmaku) [reply cloud/terraform/gcloud_config_builder.py:87]: No, not sure really, but we can also specify the instance type in the resources.json. The same for AZ and AWS. so far, this default instance type has worked well, although only 2 images have been tested. I think we will discover if this is an issue in the future.

Comment (jabia99) [reply cloud/terraform/gcloud_config_builder.py:87]: Cool, let's keep this in mind then

File: cloud/terraform/terraform_controller.py (hunk 1 above)
Comment (jabia99) [reply cloud/terraform/terraform_controller.py:126]: I don't even know why we'd want a DNS. I was just curious

**Pull request merged**